File: src/main/java/io/kenza/support/utils/Java.java
package io.kenza.support.utils;

import java.util.Optional;

public class Java {

    public static <T> Optional<T> convertInstanceOfObject(Object o, Class<T> clazz) {
        try {
            return Optional.ofNullable(clazz.cast(o));
        } catch(ClassCastException e) {
            return Optional.empty();
        }
    }
}

File: src/main/kotlin/io/kenza/support/utils/kotlin/java.kt
package io.kenza.support.utils.kotlin

import java.util.*
import java.util.stream.Collectors
import java.util.stream.Stream

val <T> Optional<T>.value: T?
    get() = orElse(null)

fun <T> Stream<T>.collectAsList(): MutableList<T> = this.collect(Collectors.toList())
fun <T> Stream<T>.collectAsSet(): MutableSet<T> = this.collect(Collectors.toSet())

